activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -23,8 +23,7 @@ def collection_cache_key(collection = all, timestamp_column = :updated_at) # :no
           query = collection.select("#{column} AS collection_cache_key_timestamp")
           subquery_alias = "subquery_for_cache_key"
           subquery_column = "#{subquery_alias}.collection_cache_key_timestamp"
-          subquery = query.arel.as(subquery_alias)
-          arel = Arel::SelectManager.new(subquery).project(select_values % subquery_column)
+          arel = query.send(:build_subquery, subquery_alias, select_values % subquery_column)
```

Comment:
I'd say make the method public, with `# :nodoc:`, but no strong options on this case.

Reply:
I'm sometimes bothered with here is on the model class scope, in spite of the implementention is for `Relation#cache_key`.

I've prepared refactoring to move the implementation from klass to relation, this makes to avoid `send` without expose the internal `build_subquery`.

https://github.com/kamipo/rails/commit/6b96d9ca2d57da075f9706d3d60c0054576bb4eb